### DISCUSSION TITLE
Does code-server with security login validate ?

### DISCUSSION THREAD

**Discussion by baoyachi** [coder/code-server#3225] [Q&A] [answered]
The `code-server` is very convenient coding tool on any mobile device. Does `code-server` with login in browser with security login validate? If yes, how to configure it.

e.g:
* ip limit
* login device limit
* secondary confirmation
* ...

**Comment by code-asher** [comment, marked as answer]
code-server only supports a single password and limits logins to two per minute plus twelve per hour.

**Reply by code-asher** [reply]
Actually we should probably document this.

**Reply by baoyachi** [reply]
I'm very interested in this. Maybe refer to this project.
[https://github.com/baoyachi/server-security](https://github.com/baoyachi/server-security)
